Code/Engine/RendererCore/Font/TextData.h

```diff
@@ -0,0 +1,360 @@
+
+    /**	Checks are we at a word boundary (meaning the next added character will start a new word). */
+    bool IsAtWordBoundary() const;
+
+    /**	Returns the total number of characters on this line. */
+    ezUInt32 GetNumCharacters() const;
+
+    /**
+			 * Query if this line was created explicitly due to a newline character. As opposed to a line that was created
+			 * because a word couldn't fit on the previous line.
+			 */
+    bool HasNewlineCharacter() const { return m_HasNewline; }
+
+  private:
+    friend class ezTextData;
+
+    /**
+			 * Appends a new character to the line.
+			 *
+			 * @param[in]	charIdx		Sequential index of the character in the original string.
+			 * @param[in]	charDesc	Character description from the font.
+			 */
+    void AddCharacter(ezUInt32 charIdx, const ezFontGlyph& charDesc);
+
+    /**	Appends a space to the line. */
+    void AddSpace(ezUInt32 spaceWidth);
+
+    /**
+			 * Adds a new word to the line.
+			 *
+			 * @param[in]	wordIdx		Sequential index of the word in the original string. Spaces are counted as words as
+			 *							well.
+			 * @param[in]	word		Description of the word.
+			 */
+    void AddWord(ezUInt32 wordIdx, const ezTextWord& word);
+
+    /** Initializes the line. Must be called after construction. */
+    void Initialize(ezTextData* textData);
+
+    /**
+			 * Finalizes the line. Do not add new characters/words after a line has been finalized.
+			 *
+			 * @param[in]	hasNewlineChar	Set to true if line was create due to an explicit newline char. As opposed to a
+			 *								line that was created because a word couldn't fit on the previous line.
+			 */
+    void Finalize(bool hasNewlineChar);
+
+    /**	Returns true if the line contains no words. */
+    bool IsEmpty() const { return m_IsEmpty; }
+
+    /**	Removes last word from the line and returns its sequential index. */
+    ezUInt32 RemoveLastWord();
+
+    /**	Calculates the line width and height in pixels. */
+    void CalculateBounds();
+
+  private:
+    ezTextData* m_TextData;
+    ezUInt32 m_WordsStart, m_WordsEnd;
+
+    ezUInt32 m_Width;
+    ezUInt32 m_Height;
+
+    bool m_IsEmpty;
+    bool m_HasNewline;
+  };
+
+public:
+  /**
+		 * Initializes a new text data using the specified string and font. Text will attempt to fit into the provided area.
+		 * If enabled it will wrap words to new line when they don't fit. Individual words will be broken into multiple
+		 * pieces if they don't fit on a single line when word break is enabled, otherwise they will be clipped. If the
+		 * specified area is zero size then the text will not be clipped or word wrapped in any way.
+		 *
+		 * After this object is constructed you may call various getter methods to get needed information.
+		 */
+  EZ_RENDERERCORE_DLL ezTextData(const ezString32& text, const ezFontResourceHandle& font, ezUInt32 fontSize, ezUInt32 width = 0, ezUInt32 height = 0, bool wordWrap = false, bool wordBreak = true);
```

Comment:
You can just mark the entire class with EZ_RENDERERCORE_DLL and don't need to do it for every function.